app/uk/gov/hmrc/cipphonenumberverification/services/NotificationService.scala

```diff
@@ -22,20 +22,25 @@ import play.api.http.Status.NOT_FOUND
 import play.api.libs.json.Json
 import play.api.mvc.Results.{NotFound, Ok}
 import play.api.mvc.{ResponseHeader, Result}
+import uk.gov.hmrc.cipphonenumberverification.audit.VerificationDeliveryResultRequestAuditEvent
 import uk.gov.hmrc.cipphonenumberverification.connectors.GovUkConnector
-import uk.gov.hmrc.cipphonenumberverification.models.{ErrorResponse, GovUkNotificationStatus, NotificationStatus}
+import uk.gov.hmrc.cipphonenumberverification.models.{AuditType, ErrorResponse, GovUkNotificationStatus, NotificationStatus}
+import uk.gov.hmrc.cipphonenumberverification.utils.GovNotifyUtils
 import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, UpstreamErrorResponse}
 
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton()
-class NotificationService @Inject()(govUkConnector: GovUkConnector)
+class NotificationService @Inject()(govNotifyUtils: GovNotifyUtils, auditService: AuditService, govUkConnector: GovUkConnector)
                                    (implicit val executionContext: ExecutionContext) extends Logging {
 
   def status(notificationId: String)(implicit hc: HeaderCarrier): Future[Result] = {
     def success(response: HttpResponse) = {
-      val (code, message) = response.json.as[GovUkNotificationStatus].status match {
+      val phoneNumber = response.json.as[GovUkNotificationStatus].phone_number
```

Comment:
might as well extract `response.json.as[GovUkNotificationStatus]` to a var

Reply:
Not sure what you mean, it is already in a var?

Reply:
`response.json.as[GovUkNotificationStatus]` isn't, and is repeated a couple more times below

Reply:
I have refactored this to use an object.  I have also refactored the unit test as they weren't testing the deserialisation.  They are now.